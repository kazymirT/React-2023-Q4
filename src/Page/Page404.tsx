import { useNavigate } from 'react-router-dom';
import { Headers } from '../Components/Headers/Headers';

export const Page404 = () => {
  const navigate = useNavigate();
  return (
    <div className="notFoundField">
      <Headers text="404" className="title" headingType="h2" />
      <p className="notFoundField__message">Page not found</p>
      <button onClick={() => navigate(`/`)}>Go to main page</button>
    </div>
  );
};
