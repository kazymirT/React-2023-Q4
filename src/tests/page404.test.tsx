import React from "react";
import { describe, expect, it, vi } from "vitest";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";

import Page404 from "../pages/404";

import mockRouter from "next-router-mock";

vi.mock("next/router", () => vi.importActual("next-router-mock"));

describe("Tests for the 404 Page component", () => {
  it("404 page is displayed when navigating to an invalid route.", async () => {
    const { getByText } = render(<Page404 />);

    expect(getByText("404")).toBeInTheDocument();
    expect(getByText("Page not found")).toBeInTheDocument();
    expect(getByText("Go to main page")).toBeInTheDocument();

    fireEvent.click(getByText("Go to main page"));
    expect(mockRouter.asPath).toBe("/");
  });
});
