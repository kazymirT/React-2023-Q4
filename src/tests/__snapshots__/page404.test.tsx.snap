// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for the 404 Page component > 404 page is displayed when navigating to an invalid route. 1`] = `
<body>
  <div>
    <div
      class="_container_fd74bb"
    >
      <h1
        class="_title_fd74bb"
      >
        404
      </h1>
      <p
        class="_message_fd74bb"
      >
        Page not found
      </p>
      <button>
        Go to main page
      </button>
    </div>
  </div>
</body>
`;
